ch32fun/ch57xhw.h

```diff
@@ -1628,6 +1633,172 @@ typedef enum
+// They should be defined in case the user is referring to the datasheet.
+#define R16_RTC_CNT_LSI     R16_RTC_CNT_32K 
+#define R16_RTC_CNT_DIV1    R16_RTC_CNT_2S 
+#define R8_LSI_CONFIG       R8_CK32K_CONFIG
+#define RB_CLK_LSI_PON      RB_CLK_XT32K_PON
+
+// Additional power plan bits.
+#define RB_PWR_LDO5V_EN     0x0100
+#define RB_PWR_RAM12K       RB_PWR_RAM2K
+#define RB_PWR_RAM24K       RB_PWR_RAM12K
+#define RB_XT_PRE_EN        0 //no such bit for ch570/2
+
+
+#define CLK_PER_US          (1.0 / ((1.0 / RTC_FREQ) * 1000 * 1000))
+#define CLK_PER_MS          (CLK_PER_US * 1000)
+#define US_TO_RTC(us)       ((uint32_t)((us) * CLK_PER_US + 0.5))
+#define MS_TO_RTC(ms)       ((uint32_t)((ms) * CLK_PER_MS + 0.5))
+
+#define SLEEP_RTC_MIN_TIME  US_TO_RTC(1000)
+#define SLEEP_RTC_MAX_TIME  (RTC_MAX_COUNT - 1000 * 1000 * 30)
+#define WAKE_UP_RTC_MAX_TIME US_TO_RTC(1600)
+
+RV_STATIC_INLINE void PFIC_EnableIRQ(IRQn_Type IRQn)
+{
+    PFIC->IENR[((uint32_t)(IRQn) >> 5)] = (1 << ((uint32_t)(IRQn) & 0x1F));
+}
+
+RV_STATIC_INLINE void LSIEnable() 
+{
+	SYS_SAFE_ACCESS(
+		R8_LSI_CONFIG |= RB_CLK_LSI_PON; //turn on LSI
+	);
+
+}
+
+// CH570/2 has no DCDC.
+#define DCDCEnable() 
+
+RV_STATIC_INLINE void SleepInit()
+{
+	SYS_SAFE_ACCESS
+    (
+		R8_RTC_MODE_CTRL |= RB_RTC_TRIG_EN;  //enable RTC trigger
+   		R8_SLP_WAKE_CTRL |= RB_SLP_RTC_WAKE; // enable wakeup control
+    );
+
+	PFIC_EnableIRQ(RTC_IRQn);//enable RTC interrupt
+
+}
+
+//clear RTC counters.
+//probabily not needed for most cases, waking up from
+//power off resets the RTC anyway.
+RV_STATIC_INLINE void RTCInit() 
+{
+	SYS_SAFE_ACCESS
+	(
+		R32_RTC_TRIG = 0;
+		R32_RTC_CTRL |= RB_RTC_LOAD_HI;
+		R32_RTC_CTRL |= RB_RTC_LOAD_LO;
+    );
+
+}
+
+RV_STATIC_INLINE void RTCTrigger(uint32_t cyc) 
+{
+	//get the rtc current time
+	uint32_t alarm = (uint32_t) R16_RTC_CNT_LSI | ( (uint32_t) R16_RTC_CNT_DIV1 << 16 );
+
+	alarm += cyc;
+
+	if( alarm > RTC_MAX_COUNT )
+	{
+		alarm-=	RTC_MAX_COUNT;
+	}
+
+    SYS_SAFE_ACCESS
+    (
+		R32_RTC_TRIG = alarm;   
+    );
+
+}
+
+// enter idle state
+RV_STATIC_INLINE void LowPowerIdle(uint32_t cyc)
+{
+	RTCTrigger(cyc);
+
+	PFIC->SCTLR &= ~(1 << 2); // don't deep sleep
+	PFIC->SCTLR &= ~(1 << 3); // wfi
+	asm volatile ("wfi\nnop\nnop" );
+
+}
+
+// This macro defines which power pin 
+// to use. If not defined correctly, sleep current
+// will be higher than expected.
+#ifndef POWERED_BY_V5PIN
+#define POWERED_BY_V5PIN 0
+#endif
+
+RV_STATIC_INLINE void LowPowerSleep(uint32_t cyc, uint16_t power_plan) 
+{
+	#if (POWERED_BY_V5PIN == 1)
+		power_plan |= RB_PWR_LDO5V_EN;
+	#endif
+
+	RTCTrigger(cyc);
+
+	#if ( CLK_SOURCE_CH5XX == CLK_SOURCE_PLL_75MHz ) || ( CLK_SOURCE_CH5XX == CLK_SOURCE_PLL_100MHz )
+		//if system clock is higher than 60Mhz, it need to be reduced before sleep.
+		SYS_SAFE_ACCESS(
+			R8_CLK_SYS_CFG = CLK_SOURCE_PLL_60MHz;
+		);
+	#endif
+
+	PFIC->SCTLR |= (1 << 2); //deep sleep
```

Comment:
This should be NVIC, to stay in line with the rest of ch32fun (there is a #define NVIC PFIC so it's just a two char change)